packages/plugins/typed-variable-modal/src/TypedVariableModal.js

```diff
@@ -0,0 +1,385 @@
+          msg = Blockly.Msg['VARIABLE_ALREADY_EXISTS'].replace(
+              '%1', existing.name);
+        } else {
+          msg =
+              Blockly.Msg['VARIABLE_ALREADY_EXISTS_FOR_ANOTHER_TYPE'];
+          msg = msg.replace('%1', existing.name).replace('%2',
+              this.getDisplayName_(existing.type));
+        }
+        Blockly.alert(msg);
+      } else {
+        // No conflict
+        this.workspace_.createVariable(text, type);
+        this.hide();
+      }
+    } else {
+      Blockly.alert(Blockly.Msg['TYPED_VAR_MODAL_INVALID_NAME']);
+    }
+  }
+
+  /**
+   * Get the display name for the given type.
+   * @param {string} type The type to get the display name for.
+   * @return {string} The display name for the type.
+   * @private
+   */
+  getDisplayName_(type) {
+    for (let i = 0; i < this.types_.length; i++) {
+      const typeNames = this.types_[i];
+      if (type === typeNames[1]) {
+        return typeNames[0];
+      }
+    }
+    return '';
+  }
+
+  /**
+   * Get the valid variable name, or null if the name is not valid.
+   * @return {string} The valid variable name, or null if the name exists.
+   * @private
+   */
+  getValidInput_() {
+    let newVar = this.variableNameInput_.value;
+    if (newVar) {
+      newVar = newVar.replace(/[\s\xa0]+/g, ' ').trim();
+      if (newVar === Blockly.Msg['RENAME_VARIABLE'] ||
+          newVar === Blockly.Msg['NEW_VARIABLE']) {
+        // Ok, not ALL names are legal...
+        newVar = null;
+      }
+    }
+    return newVar;
+  }
+
+  /**
+   * Render content for the modal content div.
+   * @param {HTMLDivElement} contentContainer The modal's content div.
+   * @override
+   */
+  renderContent_(contentContainer) {
+    const varNameContainer = this.createVarNameContainer_();
+    this.variableNameInput_ = varNameContainer
+        .querySelector('.typed-modal-variable-name-input');
+
+    const typedVarDiv = document.createElement('div');
+    Blockly.utils.dom.addClass(typedVarDiv, 'typed-modal-types');
+    typedVarDiv.innerText = Blockly.Msg["TYPED_VAR_MODAL_TYPES_LABEL"];
+
+    this.variableTypesDiv_ = this.createVariableTypeContainer_(this.types_);
+    this.resetModalInputs_();
+    typedVarDiv.appendChild(this.variableTypesDiv_);
+    contentContainer.appendChild(varNameContainer);
+    contentContainer.appendChild(typedVarDiv);
+  }
+
+  /**
+   * Render content for the modal footer.
+   * @param {HTMLElement} footerContainer The modal's footer div.
+   * @override
+   */
+  renderFooter_(footerContainer) {
+    footerContainer.appendChild(this.createConfirmBtn_());
+    footerContainer.appendChild(this.createCancelBtn_());
+  }
+
+  /**
+   * Create button in charge of creating the variable.
+   * @return {!HTMLButtonElement}The button in charge of creating a variable.
+   * @protected
+   */
+  createConfirmBtn_() {
+    const confirmBtn = document.createElement('button');
+    Blockly.utils.dom.addClass(confirmBtn, 'blockly-modal-btn');
```

Comment:
I would avoid using ``Blockly.utils.dom.addClass`` unless you have a need to add / remove a class. In this case you're creating the element right there, simply setting ``className`` to ``blockly-modal-btn  blockly-modal-btn-primary`` is clearer.